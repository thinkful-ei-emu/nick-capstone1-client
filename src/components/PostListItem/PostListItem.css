.odd-item {
  background-color:rgb(207, 55, 55);
  color: white;
  margin: 2% 2%;
  padding: 2% 0;
  border-radius: 5px;
}

.even-item {
  background-color: rgb(155, 163, 172);
  color: white;
  margin: 2% 2%;
  padding: 2% 0;
  border-radius: 5px;
}

.even-item > a {
  color: white;
}

.even-item > .toggle {
  color: white;
  background: none;
  border: none;
}

.even-item > a:hover {
  text-decoration: underline;
}

.odd-item > a:hover {
  text-decoration: underline;
}
.odd-item > a {
  color: white;
}

.odd-item > .toggle {
  color: white;
  background: none;
  border: none;
}

.toggle:focus{
  outline: none;
}

.toggle {
  background: none;
  color: black;
  border: none;
}

/*is this needed? */
.post-item {
  margin: 2% 2%; 
  padding: 2% 0;
  border-radius: 5px;
}

.item-header{
  margin: 0;
  font-size: 30px;
}



.post-item-attr {
  list-style: none;
  text-align: left;
  margin-left: 25%;
}

@media only screen and (min-width: 700px){
  .post-item-attr {
    list-style: none;
    text-align: left;
    margin-left: 30%;
  }
}


@media only screen and (min-width: 1000px){
  .odd-item {
    background-color:rgb(207, 55, 55);
    color: white;
    margin: 2% 25%;
    padding: 2% 0;
    border-radius: 5px;
  }
  .even-item {
    color: white;
    margin: 2% 25%;
    padding: 2% 0;
    border-radius: 5px;
  }
  .post-item-attr {
    list-style: none;
    text-align: left;
    margin-left: 35%;
  }
}

@media only screen and (min-width: 1300px){
  .post-item-attr {
    list-style: none;
    text-align: left;
    margin-left: 40%;
  }
}
