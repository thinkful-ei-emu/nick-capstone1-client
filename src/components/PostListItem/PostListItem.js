import React from 'react';
import { Link } from 'react-router-dom';
import './PostListItem.css';


class PostListItem extends React.Component {


  state = {
    isCollapsed: true
  }

  toggleCollapse = (ev) => {
    ev.preventDefault();
    this.setState({
      isCollapsed: !this.state.isCollapsed
    })
  }

  determineContent = (post) => {
    if(this.state.isCollapsed){
      return (
        <div>
          <Link to={`/posts/${post.id}`}>
            <h1>{post.style}</h1>
          </Link>
          <p>Location: {post.location}</p>
          <p>Need: {post.instruments_need.split(' ').join(', ')}</p>
          <p>{this.truncDesc(post.description)}</p>
        </div>
      )
    }
    else return (
      <div>
      <Link to={`/posts/${post.id}`}>
        <h1>{post.style}</h1>
      </Link>
      <h2>Posted by: <em>{post.author.user_name}</em> on {new Date(post.author.date_created).toLocaleDateString('en-US')}</h2>
      <p>Location: {post.location}</p>
      <p>Need: {post.instruments_need.split(' ').join(', ')}</p>
      <p>{post.description}</p>
      <ul>
        <li>Style: {post.style}</li>
        <li>Time: {post.commitment}</li>
        <li>Desired Skill: {post.skill_lvl}</li>
      </ul>
    </div>
    )
  }

  buttonText = () => {
    if(this.state.isCollapsed){
      return 'show more'
    }
    else return 'show less'
  }

  truncDesc = description => {
    const newDesc = description.split(' ').splice(0, 8).join(' ').concat('...')
    return newDesc;
  }

  render(){
    const { post } = this.props;
    let buttonText = this.buttonText();
    let content = this.determineContent(post);
    return (
      <div style={{ border: '1px solid red'}}>
      {content}
        <button onClick={this.toggleCollapse}>{buttonText}</button>
      </div>
    )
  }

}

export default PostListItem;


