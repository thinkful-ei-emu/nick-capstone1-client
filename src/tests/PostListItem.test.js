import React from 'react';
import ReactDOM from 'react-dom';
import PostListItem from '../components/PostListItem/PostListItem';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';




const dummyPost = {
  id: 1,
  style: 'Rock',
  location: 'Atlanta (GA)',
  instruments_need: 'Drums Piano',
  commitment: 'Low(1-3)',
  skill_lvl: 'Hobbyist',
  description: 'hello'
}

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MemoryRouter><PostListItem post={dummyPost} index={1} /></MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><PostListItem post={dummyPost} index={1}/></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
})