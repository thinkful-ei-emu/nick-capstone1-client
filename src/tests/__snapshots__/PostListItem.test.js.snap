// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="even-item"
>
  <a
    href="/posts/1"
    onClick={[Function]}
  >
    <h2
      className="item-header"
    >
      Rock
    </h2>
  </a>
  <p>
    Location: 
    Atlanta (GA)
  </p>
  <p>
    Need: 
    Drums, Piano
  </p>
  <p>
    hello...
  </p>
  <button
    className="toggle"
    onClick={[Function]}
  />
</div>
`;
